### ISSUE TITLE
more verbose json output

### ISSUE THREAD

**Issue by chmanie** [substack/clocker#25]
https://github.com/substack/clocker/pull/22 added csv output, which is great! additionally we have the `data` option which outputs json, but this is squashed to days to be compatible with invoicer.

To simplify the api I'd propose a flag for the `list` command either like this
`clocker list --output=json / --output=csv`, (or `--json` / `--csv`) because right now the `csv` command basically just reformats the `list` output. Maybe we could alias the `csv` command to not break the api?

I'd also happily start working on a PR for that. Let me know what you think.

**Comment by fnogatz** [comment]
Yeah, refactoring the different output formats is definitely a good thing to work on. I suggest the following interface:

- `clocker list`: same as today
- `clocker list --aggregate day`: aggregate by day
- `clocker list --aggregate month`: aggregate by month (and maybe other times)
- `clocker list --csv`: same as `clocker csv` today
- `clocker list --json --aggrate day`: similar to `clocker data` today
- All existing output commands are aliases to `clocker list`.

What do you think?

**Comment by yoshuawuyts** [comment]
@fnogatz lgtm!

**Comment by fnogatz** [comment]
RFC @substack

**Comment by chmanie** [comment]
I am working on some pinning tests to support the refactoring. I could share them with you soon.

**Comment by fnogatz** [comment]
Thanks for working on some tests at [chmanie/clocker](https://github.com/chmanie/clocker/tree/pinning-tests), @chmanie! My favorite approach would be:

1. Write tests to document current expected behavior
1. Create v2 branch
1. Rewrite the current features in a `/lib`, so one could also `require('clocker').stop()` etc.
1. Rewrite the current CLI to use the lib
1. Publish v2 :)
